components/filter-field/src/filter-field-default-data-source.ts

```diff
@@ -31,12 +31,15 @@ import {
 } from './types';
 
 /** Shape of an object to be usable as a option in an autocomplete */
-export type DtFilterFieldDefaultDataSourceOption = { name: string } | string;
+export type DtFilterFieldDefaultDataSourceOption =
+  | { name: string; id?: string }
```

Comment:
This looks fine to me, a comment about why we are doing this would be cool.

Reply:
I am gonna go with any since the id might be an enum / number / whatever